### PR TITLE
refactor: Add function to make all column ids in a tree unique and sequential

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/dataframe.py [hunk 1]
```diff
@@ -3472,6 +3472,52 @@ def _apply_unary_op(self, operation: ops.UnaryOp) -> DataFrame:
         block = self._block.multi_apply_unary_op(self._block.value_columns, operation)
         return DataFrame(block)
 
+    def f(
```
Comment (chelsea-lin) [inline bigframes/dataframe.py:3475]: IIUC, it's internal function. Maybe `_f` or other more readable name.

Comment (chelsea-lin) [reply bigframes/dataframe.py:3475]: I didn't see this method gets called. Please double check.

**Review by chelsea-lin — CHANGES_REQUESTED** [review]
Could you please add some tests to increase the test coverages on these core files. Thanks!
The following link is the current coverage status:
https://fusion2.corp.google.com/ci/kokoro/prod:bigframes%2Fpresubmit%2Fpresubmit/activity/88a8d1da-2043-4a7a-8ae8-0b91d7e51c84/log

File: bigframes/dataframe.py (hunk 1 above)
Comment (TrevorBergeron) [reply bigframes/dataframe.py:3475]: accidentally copied `_map_clustering_columns` removed in new revision

**Comment by TrevorBergeron** [comment]
> Could you please add some tests to increase the test coverages on these core files. Thanks! The following link is the current coverage status: https://fusion2.corp.google.com/ci/kokoro/prod:bigframes%2Fpresubmit%2Fpresubmit/activity/88a8d1da-2043-4a7a-8ae8-0b91d7e51c84/log

increasing code coverage is definitely a goal, however the code in this PR is covered (minus assert code that should never hit).

**Review by chelsea-lin — APPROVED** [review]

**Pull request merged**